src/payload.cpp

```diff
@@ -317,6 +317,19 @@ void PayloadConvert::addByte(uint8_t value) {
   not implemented
   */ }
 
+void PayloadConvert::addSDS(sdsStatus_t sds) {
+#if (HAS_SDS011)
+#if (PAYLOAD_ENCODER == 3) // Cayenne LPP dynamic
+    buffer[cursor++] = LPP_PARTMATTER_CHANNEL;    // for PM10 and PM25
+#endif
+    buffer[cursor++] = LPP_LUMINOSITY; // workaround since cayenne has no data type meter
+    buffer[cursor++] = highByte(sds.pm10);
+    buffer[cursor++] = lowByte(sds.pm10);
+    buffer[cursor++] = highByte(sds.pm25);
```

Comment:
This won't work, since LPP_LUMINOSITY takes only 2 bytes, not 4. See Cayenne LPP specs.
But you could define two channels, one for PM10 and one for PM25 value.

Please install mydevices Cayenne app on your smartphone and test your code with your device.